### DISCUSSION TITLE
18KB RAM (16K+2K just below 32K) booting PC DOS

### DISCUSSION THREAD

**Discussion by Torinde** [86Box/86Box#4548] [Q&A] [answered]
When using 8088:
- with 32KB RAM I can both boot into Cassette BASIC and into PC DOS 1.00
- with 16KB RAM I can boot only into Cassette BASIC. When DOS image is mounted there's only a blinking cursor.

How can I boot DOS with 16KB?

On real PC - IBM DOS 1.0 should [boot with 16KB](https://www.minuszerodegrees.net/5150/ram/5150_not_enough_ram_for_DOS.htm) according to this:
> Motherboard configured for 16 KB RAM
Symptom / Result: 'PARITY CHECK 1' on-screen, plus floppy drive access LED on continuously, plus floppy drive spindle turning continuously. (See note 1 below.)
Note 1: There was a successful boot if all I did was simply populate the second RAM bank, leaving the motherboard switches set to 16 KB.

But if the second RAM bank is populated - BIOS/DOS are using only 16KB (e.g. as set by the switches), while still being able to boot from 0x7C00 (which is beyond 16KB) - I assume, due to the [permanently connected U12 memory transievers](https://www.minuszerodegrees.net/5150/motherboard/IBM%205150%20motherboard%20-%20Removing%20RAM%20chips.png), which it seems would allow a quasi-16KB/32KB mode when a second bank of RAM is populated, while DIP switches are set for 16KB.

Related: https://github.com/joncampbell123/dosbox-x/issues/3664, https://github.com/640-KB/GLaBIOS/discussions/50, #4772, https://github.com/dbalsom/martypc/discussions/126#discussioncomment-9849151

PS. 86Box doesn't show 'PARITY CHECK 1' on-screen when booting PC DOS 1.00 at 16KB [because the RAM contents arn't random on 86box](https://github.com/86Box/86Box/discussions/4772#discussioncomment-12222862). 

![image](https://github.com/86Box/86Box/assets/97228894/c8a65ebf-66d4-41ac-a0dc-6790ee42b90f)

**Comment by pankoza2-pl** [comment]
If you read correctly, it did not boot on 16K, DOS 1.0 requires at least 32K

**Reply by Torinde** [reply]
Yes and no - [here](https://github.com/86Box/86Box/discussions/4772#discussioncomment-12223230)'s a discussion about it.

I suggest adding "18KB" RAM in the dropdown - special case, emulating IBM PC behavior with 16KB (soldered on motherboard) and one extra memory chip at the top of 32KB addresses (0x7800-0x7FFF), value to be reported/assigned via BIOS/POST/DOS to be 16KB (corresponding to the DIP switches on the motherboard). Scenario: "User didn't have enough RAM chips or some of them failed".

**Reply by jriwanek** [reply, marked as answer]
We will not be doing this, as covered by your similar requests elsewhere, this isn't even a configuration any PC had.

**Reply by Torinde** [reply]
OK, I understand it's an obscure detail and I agree it wasn't commonly encountered - although [as reported](https://www.minuszerodegrees.net/5150/ram/5150_not_enough_ram_for_DOS.htm) it was encountered at least once.

If it's easy to "misreport" the memory size - can you please point me to what I should change in the sources? (although, I assume it won't be so easy as probably you have only a single variable for memory size instead of two - actual and reported)

**Reply by Torinde** [reply]
PCE with 16K+7C00 PC - PC DOS 1.00:

![Image](https://github.com/user-attachments/assets/1fd55191-e574-4989-9124-f9446f049df9)
